### DISCUSSION TITLE
ParaView examples don't work on Windows

### DISCUSSION THREAD

**Discussion by pazars** [Kitware/trame#99] [Q&A]
Hi!

As the title says, I cannot get the Trame _07_paraview_ examples to work. I am using the same Python version for the virtual environment as is used by ParaView.

Could it be that the venv activation is hard-coded to bin/activate? Because my venv structure is different and is activated using Scripts/activate.bat. When running the examples, a ModuleNotFound error is raised at the first trame import. Of course Trame is installed in the venv.

**Comment by jourdain** [comment]
I guess we will need to investigate. FYI @psavery

**Reply by psavery** [reply]
@jourdain I will look into it.

**Comment by rodrigomologni** [comment]
Hi @Pazars!

Please try to replace

```python
python_libs = os.path.join(VENV_BASE, f"lib/python{sys.version_info.major}.{sys.version_info.minor}/site-packages")
```

by

```python
python_libs = os.path.join(VENV_BASE, "Lib/site-packages")
```

in

```
C:\Program Files\ParaView 5.10.1-Windows-Python3.9-msvc2017-AMD64\bin\Lib\site-packages\paraview\web\venv.py
```

Here it worked!

**Reply by psavery** [reply]
@jourdain It looks like @rodrigomologni is right. The `venv.py` file in ParaView assumes that the python libs will be located at `os.path.join(VENV_BASE, f"lib/python{sys.version_info.major}.{sys.version_info.minor}/site-packages")`, but on Windows, it looks like they are located at `python_libs = os.path.join(VENV_BASE, "Lib/site-packages")` instead.

So, we probably need to fix that [here](https://gitlab.kitware.com/paraview/paraview/-/blob/master/Web/Python/paraview/web/venv.py).

**Comment by psavery** [comment]
This issue should be fixed via [ParaView MR !5844](https://gitlab.kitware.com/paraview/paraview/-/merge_requests/5844).

Until it is in a release, I would recommend patching ParaView's `venv.py` file locally with the suggestion from @rodrigomologni .

**Reply by psavery** [reply]
This is now in ParaView master, and should be in ParaView 5.11, which is going to be released sometime soon.
